test/controllers/sections/items/ConfirmCommodityCodeControllerSpec.scala

```diff
@@ -0,0 +1,132 @@
+      .set(ItemExciseProductCodePage(testIndex1), testExciseProductCodeB000)
+      .set(ItemCommodityCodePage(testIndex1), testCommodityCodeBeer)
+
+    val confirmCodesList: Seq[SummaryListRow] = Seq(
+      ItemExciseProductCodeSummary.row(idx = testIndex1),
+      ItemCommodityCodeSummary.row(idx = testIndex1, goodsType = GoodsTypeModel(testExciseProductCodeB000.code), filledUserAnswers)).flatten
+
+    val confirmCodesSummaryList: SummaryList = SummaryListViewModel(
+      rows = confirmCodesList
+    ).withCssClass("govuk-!-margin-bottom-9")
+
+
+    lazy val errorHandler: ErrorHandler = app.injector.instanceOf[ErrorHandler]
+    val view: ConfirmCommodityCodeView = app.injector.instanceOf[ConfirmCommodityCodeView]
+  }
+
+  "ConfirmCommodityCode Controller" - {
+    ".onPageLoad" - {
+
+      "must return OK and the correct view when supplied EPC and Commodity Code" in new Fixture(
+        Some(emptyUserAnswers
+          .set(ItemExciseProductCodePage(testIndex1), testExciseProductCodeB000)
+          .set(ItemCommodityCodePage(testIndex1), testCommodityCodeBeer))) {
+
+          MockConfirmCommodityCodeHelper.summaryList(testIndex1, GoodsTypeModel(testExciseProductCodeB000.code),
+            emptyUserAnswers
+              .set(ItemExciseProductCodePage(testIndex1), testExciseProductCodeB000)
+              .set(ItemCommodityCodePage(testIndex1), testCommodityCodeBeer)
+          ).returns(confirmCodesSummaryList)
+
+          val result = controller.onPageLoad(testErn, testDraftId, testIndex1, NormalMode)(request)
+
+          val viewAsString = view(controllers.sections.items.routes.ConfirmCommodityCodeController.onSubmit(testErn, testDraftId, testIndex1, NormalMode),
+            confirmCodesSummaryList
+          )(dataRequest(request), messages(request)).toString
+
+          status(result) mustBe OK
+          contentAsString(result) mustBe viewAsString
+      }
+      "must redirect to Journey Recovery if no existing data is found" in new Fixture(None) {
+
+          val result = controller.onPageLoad(testErn, testDraftId, testIndex1, NormalMode)(request)
+
+          status(result) mustBe SEE_OTHER
+          redirectLocation(result).value mustBe routes.JourneyRecoveryController.onPageLoad().url
+
+      }
+    }
+
+    ".onSubmit" - {
+      "must redirect to the onward route" in new Fixture(Some(emptyUserAnswers
+        .set(ItemExciseProductCodePage(testIndex1), testExciseProductCodeB000)
+        .set(ItemCommodityCodePage(testIndex1), testCommodityCodeBeer))) {
+
+        def request: FakeRequest[AnyContentAsEmpty.type] =
+          FakeRequest(POST, controllers.sections.items.routes.ConfirmCommodityCodeController.onSubmit(testErn, testDraftId, testIndex1, NormalMode).url)
+
+
+          val result = controller.onSubmit(testErn, testDraftId, testIndex1, NormalMode)(request)
+
+          status(result) mustBe SEE_OTHER
+          redirectLocation(result).value mustBe testOnwardRoute.url
+
+      }
```

Comment:
Could you add a test for redirecting to Journey Recover when no data is found and indexing (out of bounds)?